File: accounts/src/main/java/br/com/netlit/accounts/infra/web/EndpointsListener.java
package br.com.netlit.accounts.infra.web;

import lombok.extern.java.Log;
import org.springframework.context.ApplicationContext;
import org.springframework.context.event.ContextRefreshedEvent;
import org.springframework.context.event.EventListener;
import org.springframework.stereotype.Component;
import org.springframework.web.method.HandlerMethod;
import org.springframework.web.servlet.mvc.method.RequestMappingInfo;
import org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerMapping;

@Log
@Component
public class EndpointsListener {

  @EventListener
  public void onApplicationEvent(final ContextRefreshedEvent event) {
    final ApplicationContext context = event.getApplicationContext();
    context.getBean(RequestMappingHandlerMapping.class).getHandlerMethods().forEach(this::logEndpoint);
  }

  private void logEndpoint(final RequestMappingInfo requestMappingInfo, final HandlerMethod handlerMethod) {
    log.info(requestMappingInfo.toString());
  }
}

File: payments/src/main/kotlin/netlit/payments/infra/web/EndpointsListener.kt
package netlit.payments.infra.web

import org.apache.logging.log4j.LogManager
import org.springframework.context.event.ContextRefreshedEvent
import org.springframework.context.event.EventListener
import org.springframework.stereotype.Component
import org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerMapping

@Component
class EndpointsListener {

    private val log = LogManager.getLogger(EndpointsListener::class.java)

    @EventListener
    fun onApplicationEvent(event: ContextRefreshedEvent) {
        val context = event.applicationContext
        val bean = context.getBean(RequestMappingHandlerMapping::class.java)
        val handlerMethods = bean.handlerMethods
        handlerMethods.forEach { (mapping, _) -> log.info(mapping) }
    }
}
